x/concentrated-liquidity/swapstrategy/one_for_zero_test.go

```diff
@@ -63,91 +62,148 @@ func (suite *StrategyTestSuite) TestComputeSwapStepOutGivenIn_OneForZero() {
+			sqrtPriceTarget:  sdk.MustNewDecFromStr("70.710663976517714496"),
+			liquidity:        sdk.MustNewDecFromStr("412478955692135.521499519343199632"),
+
+			amountOneInRemaining: sdk.NewDec(5416667230),
+			spreadFactor:         sdk.ZeroDec(),
+
+			expectedSqrtPriceNext: osmomath.MustNewDecFromStr("70.710663976517714496"),
+
+			expectedAmountInConsumed:        sdk.ZeroDec(),
+			expectedAmountOut:               sdk.ZeroDec(),
+			expectedSpreadRewardChargeTotal: sdk.ZeroDec(),
+		},
+		"7: invalid zero difference between sqrt price current and sqrt price next due to precision loss, full amount remaining in is charged and amount out calculated from sqrt price": {
+			// Note the numbers are hand-picked to reproduce this specific case.
+			sqrtPriceCurrent: osmomath.BigDecFromSDKDec(sdk.MustNewDecFromStr("0.000001000049998750")),
+			sqrtPriceTarget:  sdk.MustNewDecFromStr("0.000001000049998751"),
+			liquidity:        sdk.MustNewDecFromStr("100002498062401598791.937822606808718081"),
+
+			amountOneInRemaining: sdk.NewDec(99),
+			spreadFactor:         sdk.ZeroDec(),
+
+			// computed with x/concentrated-liquidity/python/clmath.py
+			// sqrtPriceCurrent + token_in / liquidity
+			expectedSqrtPriceNext: osmomath.MustNewDecFromStr("0.0000010000499987509899752698"),
+
+			expectedAmountInConsumed: sdk.NewDec(99),
+			// liquidity * (sqrtPriceNext - sqrtPriceCurrent) / (sqrtPriceNext * sqrtPriceCurrent)
+			// calculated with x/concentrated-liquidity/python/clmath.py
+			// diff = (sqrtPriceNext - sqrtPriceCurrent)
+			// diff = round_decimal(diff, 36, ROUND_FLOOR) (0.000000000000000000989975269800000000)
+			// mul = (sqrtPriceNext * sqrtPriceCurrent)
+			// mul = round_decimal(mul, 36, ROUND_CEILING) (0.000000000001000100000000865026329827)
+			//  round_decimal(liquidity * diff / mul, 36, ROUND_FLOOR)
+			expectedAmountOut:               osmomath.MustNewDecFromStr("98990100989815.389417309844929293132374729779331247").SDKDec(),
+			expectedSpreadRewardChargeTotal: sdk.ZeroDec(),
+		},
+		"8: invalid zero difference between sqrt price current and sqrt price next due to precision loss. Returns 0 for amounts out. Note that the caller should detect this and fail.": {
+			// Note the numbers are hand-picked to reproduce this specific case.
+			sqrtPriceCurrent: osmomath.BigDecFromSDKDec(types.MaxSqrtPrice).Sub(osmomath.SmallestDec()),
+			sqrtPriceTarget:  types.MaxSqrtPrice,
+			liquidity:        sdk.MustNewDecFromStr("100002498062401598791.937822606808718081"),
+
+			amountOneInRemaining: sdk.SmallestDec(),
+			spreadFactor:         sdk.ZeroDec(),
+
+			expectedSqrtPriceNext: types.MaxSqrtPriceBigDec.Sub(osmomath.SmallestDec()),
+
+			// Note, this case would lead to an infinite loop or no progress made in swaps.
+			// As a result, the caller should detect this and fail.
+			expectedAmountInConsumed:        sdk.ZeroDec(),
+			expectedAmountOut:               sdk.ZeroDec(),
+			expectedSpreadRewardChargeTotal: sdk.ZeroDec(),
+		},
```

Comment:
Should we not be expecting errors in either case? If not (and we confirmed there are no error cases we want to test), can we remove expected error from the struct?

Reply:
removed